server/sonar-server/src/main/java/org/sonar/server/dashboard/db/WidgetPropertyDao.java

```diff
@@ -42,4 +49,8 @@ protected WidgetPropertyDto doGetNullableByKey(DbSession session, Long propertyI
     return mapper(session).selectByDashboard(dashboardKey);
   }
 
+  public void deleteByWidgetIds(DbSession session, List<Long> widgetIdsWithPropertiesToDelete) {
+    mapper(session).deleteByWidgetIds(widgetIdsWithPropertiesToDelete);
```

Comment:
If the number of ids is more than 1000, it will fail on Oracle. 
Please use DaoUtils.executeLargeInputs() to prevent such issues.